tests/test_ply.cpp

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file point_cloud_registration.cpp
+ * @brief Point cloud registration using Iterative Closest Point (ICP) algorithm.
+ *
+ * This program implements both single-stage and multi-stage ICP for 3D point cloud
+ * registration. It supports PLY file format for input and output point clouds.
+ */
+
+#include <cstddef>
```

Comment:
Can we make this its own thing like `vis`? We can just put it under a folder `ply_vis` or something. You can also feel free to extend it with your actual visualization code?